src/estop.cpp

```diff
@@ -37,6 +41,26 @@ void sigintShutdownNode(int sig)
 {
     gpioSetValue(ESTOP_PIN, 0);
     ros::shutdown();
+    g_request_shutdown = 1; // Set flag
+}
+
+/**
+ * This function is called before the node is killed using rosnode kill. It disables the estop and shutdown the node. 
+*/
+void shutdownCallback(XmlRpc::XmlRpcValue& params, XmlRpc::XmlRpcValue& result)
+{
+  int num_params = 0;
+  if (params.getType() == XmlRpc::XmlRpcValue::TypeArray)
+    num_params = params.size();
+  if (num_params > 1)
```

Comment:
Instead of add num_params & etc, could we just use params.size() in the if condition ?

Reply:
That does function properly on my pc, but XmlRpc::XmlRpcValue::TypeArray is from an enum containing 6 other possible types. As such, params could theoretically also be one of those type. I have no clue in what scenario that would happen though.

Reply:
Ah I see. That's fair. Thanks for clearing things up for me.